src/input/pointer.js

```diff
@@ -243,63 +245,136 @@
      */
     function dispatchEvent(e) {
         var handled = false;
-        var handlers = evtHandlers[e.type];
-
-        // Convert touchcancel -> touchend, and pointercancel -> pointerup
-        if (!handlers) {
-            if (activeEventList.indexOf(e.type) === POINTER_CANCEL) {
-                handlers = evtHandlers[activeEventList[POINTER_UP]];
-            } else {
-                handlers = evtHandlers[e.type];
-            }
-        }
-
-        if (handlers) {
-            // get the current screen to world offset
-            me.game.viewport.localToWorld(0, 0, viewportOffset);
-            for (var t = 0, l = changedTouches.length; t < l; t++) {
-                // Do not fire older events
-                if (typeof(e.timeStamp) !== "undefined") {
-                    if (e.timeStamp < lastTimeStamp) {
-                        continue;
+        
+        for (var guid in evtHandlers) {
+            if (evtHandlers.hasOwnProperty(guid)) {
```

Comment:
@parasyte: doesn't creating an anonymous function everytime adds overhead?
@obiot: testing has been done on Chrome and CocoonJS for Android.